src/main/java/com/checkmarx/flow/custom/GitLabIssueTracker.java

```diff
@@ -71,19 +70,18 @@ public void init(ScanRequest request, ScanResults results) throws MachinaExcepti
         }
     }
 
-    private Integer getProjectDetails(String repoName) {
+    private Integer getProjectId(String targetNamespace, String targetRepoName) {
         try {
-            String url = properties.getApiUrl().concat(PROJECT);
-            url = url.replace("{repo}", repoName);
-            URI uri = new URI(url);
-            HttpEntity httpEntity = new HttpEntity<>(createAuthHeaders());
-            ResponseEntity<String> response = restTemplate.exchange(uri, HttpMethod.GET, httpEntity, String.class);
-            JSONArray arr = new JSONArray(response.getBody());
-            for(Object obj: arr){
-                JSONObject realObj = (JSONObject) obj;
-                String name = realObj.getString("name");
-                if(name.equals(repoName)) {
-                    return realObj.getInt("id");
+            JSONArray candidateProjects = getProjectSearchResults(targetRepoName);
```

Comment:
why don't you fix this function, to use the namespace in the query?
this will remove all the loop logic here

Reply:
I wasn't able to find a suitable API for such search.
https://docs.gitlab.com/ee/api/projects.html#search-for-projects-by-name - here the query is "A string contained in the project name", so for exact match the loop will still be required.

https://docs.gitlab.com/ee/api/projects.html#list-all-projects - there is a `search_namespaces` parameter, but again this is not our case, according to the docs.